src/main/java/com/odysseusinc/athena/service/impl/NotificationServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+package com.odysseusinc.athena.service.impl;
+
+import com.odysseusinc.athena.model.athena.Notification;
+import com.odysseusinc.athena.model.athena.VocabularyConversion;
+import com.odysseusinc.athena.model.athenav5.VocabularyV5;
+import com.odysseusinc.athena.repositories.athena.NotificationRepository;
+import com.odysseusinc.athena.repositories.athena.VocabularyConversionRepository;
+import com.odysseusinc.athena.repositories.v5.VocabularyRepository;
+import com.odysseusinc.athena.service.NotificationService;
+import com.odysseusinc.athena.service.mail.EmailService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.commons.lang.StringUtils.equalsIgnoreCase;
+
+@Service
+@Transactional
+public class NotificationServiceImpl implements NotificationService {
+
+    private final EmailService emailService;
+    private final NotificationRepository notificationRepository;
+    private final UserService userService;
+    private final VocabularyConversionRepository vocabularyConversionRepository;
+    private final VocabularyRepository vocabularyRepository;
+
+    @Autowired
+    public NotificationServiceImpl(EmailService emailService, NotificationRepository notificationRepository, UserService userService, VocabularyConversionRepository vocabularyConversionRepository, VocabularyRepository vocabularyRepository) {
+        this.emailService = emailService;
+        this.notificationRepository = notificationRepository;
+        this.userService = userService;
+        this.vocabularyConversionRepository = vocabularyConversionRepository;
+        this.vocabularyRepository = vocabularyRepository;
+    }
+
+    @Override
+    public void createNotificationSubscriptions(Long userId, String[] vocabularyCodes) {
```

Comment:
Not an issue but... It's already have 'notification' noun in the class name, is it necessary to duplicate one in every method name? 
Does NotificationService work with something over than notification?

Reply:
changed